pkg/clients/cache.go

```diff
@@ -168,192 +159,128 @@ func NewSessionCache(cfg config.ScrapeConf, fips bool, logger logging.Logger) Se
-			s.clients[role][region].ec2 = createEC2Session(s.session, &region, role, s.fips, s.logger.IsDebugEnabled())
-			s.clients[role][region].dms = createDMSSession(s.session, &region, role, s.fips, s.logger.IsDebugEnabled())
-			s.clients[role][region].apiGateway = createAPIGatewaySession(s.session, &region, role, s.fips, s.logger.IsDebugEnabled())
-			s.clients[role][region].storageGateway = createStorageGatewaySession(s.session, &region, role, s.fips, s.logger.IsDebugEnabled())
-			s.clients[role][region].prometheus = createPrometheusSession(s.session, &region, role, s.fips, s.logger.IsDebugEnabled())
+			cachedClient.tagging = createTaggingClient(c.session, &region, role, c.fips, c.logger)
+			cachedClient.account = createAccountClient(c.logger, c.stscache[role])
 		}
 	}
 
-	s.cleared = false
-	s.refreshed = true
-}
-
-func (s *sessionCache) GetSTS(role config.Role) stsiface.STSAPI {
-	// if we have not refreshed then we need to lock in case we are accessing concurrently
-	if !s.refreshed {
-		s.mu.Lock()
-		defer s.mu.Unlock()
-	}
-	if sess, ok := s.stscache[role]; ok && sess != nil {
-		return sess
-	}
-	s.stscache[role] = createStsSession(s.session, role, s.stsRegion, s.fips, s.logger.IsDebugEnabled())
-	return s.stscache[role]
-}
-
-func (s *sessionCache) GetCloudwatch(region *string, role config.Role) cloudwatchiface.CloudWatchAPI {
-	// if we have not refreshed then we need to lock in case we are accessing concurrently
-	if !s.refreshed {
-		s.mu.Lock()
-		defer s.mu.Unlock()
-	}
-	if sess, ok := s.clients[role][*region]; ok && sess.cloudwatch != nil {
-		return sess.cloudwatch
-	}
-	s.clients[role][*region].cloudwatch = createCloudwatchSession(s.session, region, role, s.fips, s.logger.IsDebugEnabled())
-	return s.clients[role][*region].cloudwatch
-}
-
-func (s *sessionCache) GetTagging(region *string, role config.Role) resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI {
-	// if we have not refreshed then we need to lock in case we are accessing concurrently
-	if !s.refreshed {
-		s.mu.Lock()
-		defer s.mu.Unlock()
-	}
-	if sess, ok := s.clients[role][*region]; ok && sess.tagging != nil {
-		return sess.tagging
-	}
-
-	s.clients[role][*region].tagging = createTagSession(s.session, region, role, s.fips)
-	return s.clients[role][*region].tagging
+	c.cleared = false
+	c.refreshed = true
 }
 
-func (s *sessionCache) GetASG(region *string, role config.Role) autoscalingiface.AutoScalingAPI {
-	// if we have not refreshed then we need to lock in case we are accessing concurrently
-	if !s.refreshed {
-		s.mu.Lock()
-		defer s.mu.Unlock()
-	}
-	if sess, ok := s.clients[role][*region]; ok && sess.asg != nil {
-		return sess.asg
-	}
-
-	s.clients[role][*region].asg = createASGSession(s.session, region, role, s.fips)
-	return s.clients[role][*region].asg
+func createCloudWatchClient(s *session.Session, region *string, role config.Role, fips bool, logger logging.Logger) cloudwatch_client.Client {
```

Comment:
nit: in other parts of the code `logger` is usually the first param, or the second in case `ctx` is passed as well.